spinn_front_end_common/interface/interface_functions/application_finisher.py

```diff
@@ -26,7 +26,7 @@ def __call__(self, app_id, txrx, executable_types):
 
         # check that the right number of processors are finished
         processors_finished = txrx.get_core_state_count(
-            app_id, CPUState.FINISHED)
+            app_id, ExecutableType.USES_SIMULATION_INTERFACE.end_state[0])
```

Comment:
Why only 0?  Should it add up all possible end_states?

Reply:
well there's only one finished state for simulation interface. you could put a iterator on there, and add them all up and aggregate the total. But seemed a bit over kill (plus i was trying to figure out why the thing was blowing up in the first place. lol. get_core_state_count doesn't like lists it seems. lol

Reply:
I think the option is either to do the iteration or make the transceiver version accept a list.  The issue is protection against future changes - if we change this so it has more than one state, this bit will go wrong.

The simple solution is to use the python sum construct:
```
processors_finished = sum(
    txrx.get_core_state_count(app_id, state) 
    for state in ExecutableType.USES_SIMULATION_INTERFACE.end_state)
```

Reply:
fixed as requested